gammapy/modeling/models/core.py

```diff
@@ -361,6 +362,46 @@ def to_dict(self):
         else:
             return {"components": models_data}
 
+    def to_parameters_table(self):
+        """Convert Models parameters to an astropy Table."""
+        pars = self.parameters.to_table()
+        rows = []
+        for name in self.parameters_unique_names:
+            row = {}
+            #Warning: splitting of parameters will break is source name has a "." in its name.
```

Comment:
Even for `BackgroundModel` we now have the `spectral` prefix in the parameter name. So you can actually rely on the order of the string components in the unique name like `model.type.name`. However we should add a check and disallow parameter names with a `.` in it...this is a change that I can do.

Reply:
there was an issue before of length 2 or 3 when `BackgroundModel` or other param. This is now simplified.
Avoiding a `.` in the name of the source should be added to prevent this from braking.

Reply:
I thing this is fixed (except the fact to add a check to prevent `.` in names.